src/main.rs

```diff
@@ -406,11 +407,23 @@ fn main() {
     let source_config = OperatorConfig::new().name("SourceOperator");
     let source_stream = erdos::connect_source(SourceOperator::new, || {}, source_config);
 
-    let square_config = OperatorConfig::new().name("SquareOperator");
-    let square_stream = erdos::connect_one_in_one_out(
-        SquareOperator::new,
-        SquareOperatorState::new,
-        square_config,
+    // let square_config = OperatorConfig::new().name("SquareOperator");
+    // let square_stream = erdos::connect_one_in_one_out(
+    //     SquareOperator::new,
+    //     SquareOperatorState::new,
+    //     square_config,
+    //     &source_stream,
+    // );
+
+    fn square_fn(a: &usize) -> usize {
+        return a * a;
+    }
+
+    let map_config = OperatorConfig::new().name("MapOperator");
+    let map_stream = erdos::connect_one_in_one_out(
+        || -> MapOperator<usize, usize> { MapOperator::new(square_fn) },
+        || {},
+        map_config,
```

Comment:
Instead of replacing the `SquareOperator`, can you please implement a `MapOperator` that doubles the value before sending it to `SquareOperator`? We have a deadline example on `SquareOperator` that we would like to keep in `main.rs`.